WcaOnRails/app/models/event.rb

```diff
@@ -0,0 +1,321 @@
+    },
+    {
+      id: "555bf",
+      name: "5x5 Cube: Blindfolded",
+      rank: 510,
+      format: "time",
+      cellName: "5x5 blindfolded",
+    },
+    {
+      id: "333mbf",
+      name: "Rubik's Cube: Multiple Blindfolded",
+      rank: 520,
+      format: "multi",
+      cellName: "3x3 multi blind",
+    },
+    {
+      id: "magic",
+      name: "Rubik's Magic",
+      rank: 997,
+      format: "time",
+      cellName: "Rubik's Magic",
+    },
+    {
+      id: "mmagic",
+      name: "Master Magic",
+      rank: 998,
+      format: "time",
+      cellName: "Master Magic",
+    },
+    {
+      id: "333mbo",
+      name: "Rubik's Cube: Multi blind old style",
+      rank: 999,
+      format: "multi",
+      cellName: "3x3 multi blind old",
+    },
+    {
+      id: "333ni",
+      name: "Rubik's Cube: No inspection",
+      rank: 1010,
+      format: "time",
+      cellName: "3x3 no inspection",
+    },
+    {
+      id: "333sbf",
+      name: "Rubik's Cube: Speed Blindfolded",
+      rank: 1020,
+      format: "time",
+      cellName: "3x3 speed blindfolded",
+    },
+    {
+      id: "333r3",
+      name: "Rubik's Cube: 3 in a row",
+      rank: 1030,
+      format: "time",
+      cellName: "3x3 3 in a row",
+    },
+    {
+      id: "333ts",
+      name: "Rubik's Cube: Team solving",
+      rank: 1035,
+      format: "time",
+      cellName: "3x3 team solving",
+    },
+    {
+      id: "333bts",
+      name: "Rubik's Cube: Blindfolded team solving",
+      rank: 1040,
+      format: "time",
+      cellName: "3x3 blindfolded team solving",
+    },
+    {
+      id: "222bf",
+      name: "2x2 Cube: Blindfolded",
+      rank: 1050,
+      format: "time",
+      cellName: "2x2 blindfolded",
+    },
+    {
+      id: "clkbf",
+      name: "Rubik's Clock: Blindfolded",
+      rank: 1060,
+      format: "time",
+      cellName: "Clock blindfolded",
+    },
+    {
+      id: "333si",
+      name: "Siamese Rubik's Cube",
+      rank: 1100,
+      format: "time",
+      cellName: "Siamese Cube",
+    },
+    {
+      id: "rainb",
+      name: "Rainbow Cube",
+      rank: 1110,
+      format: "time",
+      cellName: "Rainbow Cube",
+    },
+    {
+      id: "snake",
+      name: "Rubik's Snake",
+      rank: 1120,
+      format: "time",
+      cellName: "Rubik's Snake",
+    },
+    {
+      id: "mirbl",
+      name: "Mirror Blocks",
+      rank: 1140,
+      format: "time",
+      cellName: "Mirror Blocks",
+    },
+    {
+      id: "360",
+      name: "Rubik's 360",
+      rank: 1150,
+      format: "time",
+      cellName: "Rubik's 360",
+    },
+    {
+      id: "222oh",
+      name: "2x2 Cube: One-handed",
+      rank: 1200,
+      format: "time",
+      cellName: "2x2 one-handed",
+    },
+    {
+      id: "magico",
+      name: "Rubik's Magic: One-handed",
+      rank: 1210,
+      format: "time",
+      cellName: "Magic one-handed",
+    }
+  ].each do |event_json|
+    event_json[:valid] = true
+    Event.new(event_json)
```

Comment:
Instead of using class variables and mutating them through instantiation, you could us a constant like so:

```ruby
ALL_EVENTS = [
  { … },
  { … }
].map { |e| Event.new(e.merge(valid: true)) }
```

`find_by_id` could be either implemented using `Array#find` or by changing `ALL_EVENTS` to a HashMap (and using `Hash#values` for `.all`).